Source/misdat.h

```diff
@@ -15,6 +15,7 @@ enum mienemy_type : uint8_t {
 	TARGET_MONSTERS,
 	TARGET_PLAYERS,
 	TARGET_BOTH,
+	TARGET_NONE,

```

Comment:
I have a question about save-game compatibility:
We introduce a new enum-Value that vanilla doesn't know about.

How much do we care?

1. It must be 100% compatible. We shouldn't introduce new variables, (enum) values and we should fill all variables vanilla uses, even if DevilutionX don't use them.
2. As long as vanilla can load it, don't crash and works normal it's okay. only test it also in vanilla.
3. We only need to read vanilla saves but don't need to write compatible saves.

There are perhaps more fine granular steps between this three options.

I think this pr is in the second category and I saw other pr's with changes in this category. 
For the Animation Distribution Logic the sourcecode could also be simplified if we allowed changes with category 2.
So I'm really interested in your opinions. ☺️ 

If this discussion is out of scope of this pr, I can also open a separate issue or we can discuss it on discord. 🙂

Reply:
Disclaimer: I have no official word from the maintainers to back up this claim, but here are my thoughts on the matter.

IMO, save file compatibility is mostly important for testing purposes. It's most convenient if the game produces save files that are directly compatible with vanilla so your category 1 and 2 should both be fine. Having said that, I noticed an effort from over a year ago to use JSON for save files instead of vanilla's MPQ format (#614). This effort included a converter to ensure compatibility. This suggests that category 3 is still okay so long as there's a conversion step to enable testing.

Perhaps it's worth considering to mark changes like this one with a comment in the code, signaling that this enum value is not supported by vanilla. At the same time, I can't really make a solid case for why doing so would be at all valuable. It's already documented in this PR, and we can always reference the devilution codebase for comparison, so this is probably good enough.

Reply:
I think we are at level 2 for compatibility. Once we get automated testing we might even move the dial to level 3.

The render fixes we have dropped the flag that tracks if things are sticking through the wall, so for a few frames, things might poke through the wall if you load a DeivlutionX save in vanilla. But gameplay doesn't diverge. I'm ok with the same in terms of the animation logic. If the animation has a little visual hiccup it's ok, but preferably the game logic should result in the same state (the attack should still take the same time).